src/drawers/drawer.ts

```diff
@@ -74,7 +74,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.keyFunction()) {
```

Comment:
Noted, done. Remember that the other test (for the existence of dataset) we have already discussed.